gems/aws-sdk-core/lib/aws-sdk-core/plugins/jsonvalue_converter.rb

```diff
@@ -11,15 +11,43 @@ class Handler < Seahorse::Client::Handler
 
         def call(context)
           context.operation.input.shape.members.each do |m, ref|
-            if ref['jsonvalue']
-              param_value = context.params[m]
-              unless param_value.respond_to?(:to_json)
-                raise ArgumentError, "The value of params[#{m}] is not JSON serializable."
+            convert_jsonvalue(m, ref, context.params, 'params')
```

Comment:
Edit: This doesn't quite work - because we need to set values back on the params it gets tricky.  The code ends up being a bit cleaner here if we keep the top level iteration of input members.